### PR TITLE
Django 1.9

### PR DESCRIPTION
**Pull request by nickpell**
Opening for tests and lolz

### CODE REVIEW

**Comment by snopoke** [comment]
🏃

**Comment by dimagimon** [comment]
There are 7 errors:

* custom/apps/gsid/__init__.py, line 1 - E501 line too long (117 > 115 characters)
* custom/intrahealth/models.py, line 11 - E502 the backslash is redundant between brackets
* custom/intrahealth/models.py, line 14 - E131 continuation line unaligned for hanging indent
* custom/intrahealth/models.py, line 16 - E502 the backslash is redundant between brackets
* custom/intrahealth/models.py, line 22 - E502 the backslash is redundant between brackets
* custom/intrahealth/utils.py, line 15 - F401 'report_calcs' imported but unused
* custom/m4change/__init__.py, line 2 - E501 line too long (119 > 115 characters)

**Comment by dimagimon** [comment]
There are 37 errors:

* custom/_legacy/mvp/reports/__init__.py, line 90 - E121 continuation line under-indented for hanging indent
* custom/apps/crs_reports/reports.py, line 241 - E201 whitespace after '{'
E203 whitespace before ':'
E231 missing whitespace after ':'
* custom/apps/crs_reports/reports.py, line 244 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 247 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 250 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 253 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 256 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 259 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 262 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 265 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 268 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 271 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 274 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 277 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 278 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 279 - E201 whitespace after '{'
E203 whitespace before ':'
E231 missing whitespace after ':'
* custom/apps/crs_reports/reports.py, line 282 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 285 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 288 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 291 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 294 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 297 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 300 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 301 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 302 - E201 whitespace after '{'
E203 whitespace before ':'
E231 missing whitespace after ':'
* custom/apps/crs_reports/reports.py, line 305 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 308 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 311 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 314 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 317 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 320 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 323 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 326 - E124 closing bracket does not match visual indentation
* custom/apps/crs_reports/reports.py, line 327 - E124 closing bracket does not match visual indentation
* custom/apps/gsid/reports/__init__.py, line 8 - W191 indentation contains tabs
* custom/m4change/reports/__init__.py, line 27 - E501 line too long (117 > 115 characters)
* custom/succeed/reports/__init__.py, line 296 - E121 continuation line under-indented for hanging indent

**Comment by dimagimon** [comment]
There are 18 errors:

* corehq/apps/app_manager/urls.py, line 24 - E501 line too long (116 > 115 characters)
* corehq/apps/app_manager/urls.py, line 26 - E501 line too long (127 > 115 characters)
* corehq/apps/app_manager/urls.py, line 27 - E501 line too long (121 > 115 characters)
* corehq/apps/cloudcare/urls.py, line 40 - E261 at least two spaces before inline comment
* corehq/apps/domain/urls.py, line 87 - E501 line too long (127 > 115 characters)
* corehq/apps/domain/urls.py, line 105 - E241 multiple spaces after ','
* corehq/apps/domain/urls.py, line 106 - E127 continuation line over-indented for visual indent
* corehq/apps/domain/urls.py, line 167 - E501 line too long (118 > 115 characters)
* corehq/apps/fixtures/urls.py, line 20 - E501 line too long (117 > 115 characters)
* corehq/apps/indicators/urls.py, line 12 - E121 continuation line under-indented for hanging indent
* corehq/apps/reports/urls.py, line 107 - E501 line too long (119 > 115 characters)
* corehq/apps/reports/urls.py, line 111 - E501 line too long (127 > 115 characters)
* corehq/apps/reports/urls.py, line 113 - E501 line too long (119 > 115 characters)
* corehq/apps/reports/urls.py, line 114 - E501 line too long (134 > 115 characters)
* corehq/apps/sms/urls.py, line 42 - E501 line too long (124 > 115 characters)
* corehq/apps/users/urls.py, line 48 - E501 line too long (121 > 115 characters)
* custom/apps/crs_reports/urls.py, line 6 - E501 line too long (128 > 115 characters)
* custom/apps/crs_reports/urls.py, line 7 - E501 line too long (124 > 115 characters)

**Comment by dimagimon** [comment]
There are 18 errors:

* corehq/apps/app_manager/urls.py, line 24 - E501 line too long (116 > 115 characters)
* corehq/apps/app_manager/urls.py, line 26 - E501 line too long (127 > 115 characters)
* corehq/apps/app_manager/urls.py, line 27 - E501 line too long (121 > 115 characters)
* corehq/apps/cloudcare/urls.py, line 40 - E261 at least two spaces before inline comment
* corehq/apps/domain/urls.py, line 87 - E501 line too long (127 > 115 characters)
* corehq/apps/domain/urls.py, line 105 - E241 multiple spaces after ','
* corehq/apps/domain/urls.py, line 106 - E127 continuation line over-indented for visual indent
* corehq/apps/domain/urls.py, line 167 - E501 line too long (118 > 115 characters)
* corehq/apps/fixtures/urls.py, line 20 - E501 line too long (117 > 115 characters)
* corehq/apps/indicators/urls.py, line 12 - E121 continuation line under-indented for hanging indent
* corehq/apps/reports/urls.py, line 107 - E501 line too long (119 > 115 characters)
* corehq/apps/reports/urls.py, line 111 - E501 line too long (127 > 115 characters)
* corehq/apps/reports/urls.py, line 113 - E501 line too long (119 > 115 characters)
* corehq/apps/reports/urls.py, line 114 - E501 line too long (134 > 115 characters)
* corehq/apps/sms/urls.py, line 42 - E501 line too long (124 > 115 characters)
* corehq/apps/users/urls.py, line 48 - E501 line too long (121 > 115 characters)
* custom/apps/crs_reports/urls.py, line 6 - E501 line too long (128 > 115 characters)
* custom/apps/crs_reports/urls.py, line 7 - E501 line too long (124 > 115 characters)

File: corehq/apps/data_interfaces/tests/test_auto_case_updates.py
Comment (dimagimon) [inline corehq/apps/data_interfaces/tests/test_auto_case_updates.py]: E302 expected 2 blank lines, found 1

**Comment by emord** [comment]
nice :white_check_mark:

**Comment by nickpell** [comment]
🎉 🍻

**Comment by benrudolph** [comment]
wow :clap:

**Comment by emord** [comment]
just deployed this to staging but had to change perms to our psql user for the following tables:

```
django_admin_log
django_session
django_site
```

We'll have to do that on prod as well, but wondering if it seems worthwhile to have an ansible command to change all tables to the correct user? The reason this happens is because prod and staging were not deployed with ansible

**Comment by snopoke** [comment]
Could be useful to have in ansible though setting PG permissions is
notoriously hard

**Comment by benrudolph** [comment]
(rebased master for staging)

**Comment by nickpell** [comment]
holding off merging until can be timed with an NIC deploy

**Comment by nickpell** [comment]
Tests got broken - they previously worked [here](https://travis-ci.org/dimagi/commcare-hq/builds/182713245).

Wondering if anyone has suggestions on how to debug the [failure](https://s3.amazonaws.com/archive.travis-ci.org/jobs/190620550/log.txt)?

```
======================================================================
ERROR: test suite for <corehq.tests.nose.HqdbContext object at 0x7fb9ff33d550>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 296, in setUp
    self.setupContext(context)
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/vendor/lib/python2.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/mnt/commcare-hq-ro/corehq/tests/nose.py", line 224, in setup
    super(HqdbContext, self).setup()
  File "/vendor/lib/python2.7/site-packages/django_nose/plugin.py", line 428, in setup
    self.old_names = self.runner.setup_databases()
  File "/usr/local/lib/python2.7/site-packages/django/test/runner.py", line 482, in setup_databases
    self.parallel, **kwargs
  File "/usr/local/lib/python2.7/site-packages/django/test/runner.py", line 726, in setup_databases
    serialize=connection.settings_dict.get("TEST", {}).get("SERIALIZE", True),
  File "/usr/local/lib/python2.7/site-packages/django/db/backends/base/creation.py", line 70, in create_test_db
    run_syncdb=True,
  File "/usr/local/lib/python2.7/site-packages/django/core/management/__init__.py", line 119, in call_command
    return command.execute(*args, **defaults)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/base.py", line 399, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python2.7/site-packages/django/core/management/commands/migrate.py", line 89, in handle
    executor = MigrationExecutor(connection, self.migration_progress_callback)
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/executor.py", line 20, in __init__
    self.loader = MigrationLoader(self.connection)
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/loader.py", line 49, in __init__
    self.build_graph()
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/loader.py", line 176, in build_graph
    self.applied_migrations = recorder.applied_migrations()
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/recorder.py", line 65, in applied_migrations
    self.ensure_schema()
  File "/usr/local/lib/python2.7/site-packages/django/db/migrations/recorder.py", line 52, in ensure_schema
    if self.Migration._meta.db_table in self.connection.introspection.table_names(self.connection.cursor()):
  File "/usr/local/lib/python2.7/site-packages/django/test/testcases.py", line 154, in __call__
    "set %s.allow_database_queries to True to silence this failure." % self.cls_name
AssertionError: Database queries aren't allowed in SimpleTestCase. Either use TestCase or TransactionTestCase to ensure proper test isolation or set TestUserESAccessors.allow_database_queries to True to silence this failure.
```

It's happening in the test runner, rather an in an individual test.  I haven't been able to reproduce it locally.  I'm guessing a ```SimpleTestCase``` got introduced that did not call ```tearDown``` or ```tearDownClass```.  (maybe @millerdev has ideas?)

**Comment by millerdev** [comment]
I'm not really sure what's going on here. That's a confusing error message, mostly because it's not being thrown in the context of a `SimpleTestCase` that I can see. I wonder how django is making that determination? I'd be happy to do a pair-debugging session if you want another set of eyes to try and figure it out. Maybe we need to rebase our fork on a newer version of `django-nose`?

**Comment by nickpell** [comment]
✅ it's go time

**Comment by emord** [comment]
Incredibly small PR. Will let you coordinate with interrupt on deploy, but good from me.

**Review by millerdev — APPROVED** [review]

**Comment by millerdev** [comment]
What fixed the test runner error we were seeing earlier?

**Comment by benrudolph** [comment]
💰

**Comment by nickpell** [comment]
@millerdev https://github.com/dimagi/commcare-hq/pull/14753/files#diff-70858a111850ce0ab3e0288b828fe570 fixed the tests.

**Pull request merged**
